portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -30,39 +31,47 @@
 import java.util.List;
 import com.google.gson.Gson;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that deals with comments (POST and GET requests) */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  //private ArrayList<String> arr = new ArrayList<String>(); 
-
+  /**
+  * Get comments from Database (correct number and sorting using query strings) 
+  */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    int numComments = Integer.parseInt(request.getParameter("count"));
+    String sortingOrder = request.getParameter("sort");
+    Query query = prepareQuery(sortingOrder);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
+    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withLimit(numComments));
+    
 
     // populate array with data from the DB
     List<Comment> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
+    for (Entity entity : results) {
         long id = entity.getKey().getId();
         String name = (String) entity.getProperty("name");
         String comment_content = (String) entity.getProperty("comment");
         long timestamp = (long) entity.getProperty("timestamp");
+        long comment_length = (long) entity.getProperty("length");
 
-        Comment comment = new Comment(id, name, comment_content, timestamp);
+        Comment comment = new Comment(id, name, comment_content, timestamp, comment_length);
         comments.add(comment);
     }
 
-    // Convert arr to JSON
+    // Convert comments into readable data type for client
     String json = convertToJson(comments); 
     
     // Send response to client
     response.setContentType("application/json");
     response.getWriter().println(json);
   }
 
+  /**
+  * Put new comment in the Database with data from POST
+  */
```

Comment:
nit: spacing (https://google.github.io/styleguide/javaguide.html#s7.1.1-javadoc-multi-line)
```suggestion
  /**
   * Put new comment in the Database with data from POST
   */
```